packages/koehnlab/io/basis_util.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+
+def get_multispinmat_prod(s,spat_nums,coord):
+        """
+        Returns the spin matrix of all given Spins S in the productbasis of all spins and all spatial coordinates
+
+        Args:
+        --------------------------
+        s -- array of multiple spin states, sorted in descending order
+        spat_nums -- array of numbers of spatial states corresponding to the spin states
+        coord -- 0,1,2 for x,y,z
+
+        Returns:
+        --------------------------
+        S -- spin matrix in Productbasis of given spin and spatial states
+        """
+        multiplicities = []
+        for i in range(len(s)):
+                multiplicities.append(int(2*s[i]+1))
+        dim = int(np.sum(multiplicities*spat_nums))
+        S = np.zeros((dim,dim),dtype = complex)
+        lower_bound = 0
+        upper_bound = 0
+        for mult in multiplicities:
+                i = multiplicities.index(mult)
+                spinmat_x,spinmat_y,spinmat_z = get_spinmat(mult)
+                spinmat = [spinmat_x,spinmat_y,spinmat_z]
+                spinmat_prod = get_spinmat_prod(spinmat[coord],mult,spat_nums[i])
```

Comment:
Why are you computing all spin matrices if you only need the one for the specific coordinate?